nixos/modules/services/cluster/druid/default.nix

```
@@ -0,0 +1,225 @@
+    enable = mkEnableOption (serviceName);
+    restartIfChanged = mkOption {
+      type = types.bool;
+      description = ''
+        Automatically restart the service on config change.
+        This can be set to false to defer restarts on clusters running critical applications.
+        Please consider the security implications of inadvertently running an older version,
+        and the possibility of unexpected behavior caused by inconsistent versions across a cluster when disabling this option.
+      '';
+      default = false;
+    };
+    config = mkOption {
+      default = { };
+      type = types.attrsOf types.anything;
+      description = ''
+        (key=value) Configuration to be written to runtime.properties of the druid ${serviceName}
+        <https://druid.apache.org/docs/latest/configuration/index.html>
+      '';
+      example = {
+        "druid.plainTextPort" = "8082";
+        "druid.service" = "servicename";
+      };
+    };
+    jdk = mkPackageOptionMD pkgs "JDK" {
+      default = [ "jdk17_headless" ];
+    };
+    jvmArgs = mkOption {
+      type = types.str;
+      default = "";
+      description = "Arguments to pass to the JVM";
+    };
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open firewall ports for ${serviceName}.";
+    };
+    internalConfig = mkOption {
+      default = { };
+      type = types.attrsOf types.anything;
+      internal = true;
+      description = "Internal Option to add to runtime.properties for ${serviceName}.";
+    };
+  };
+
+
+
+  druidServiceConfig =
+    { name
+    , serviceOptions ? cfg."${name}"
+    , allowedTCPPorts ? [ ]
+    , tmpDirs ? [ ]
+    , extraConfig ? { }
+    }: (
+      mkIf serviceOptions.enable (mkMerge [
+        {
+          systemd = {
+            services."druid-${name}" = {
+              after = [ "network.target" ];
+              description = "Druid ${name}";
+              environment = {
+                DRUID_JAVA_HOME = "${serviceOptions.jdk}";
+              };
+              wantedBy = [ "multi-user.target" ];
+              inherit (serviceOptions) restartIfChanged;
+              path = [ "${serviceOptions.jdk}" ];
+              script =
+                let
+                  cfgFile = fileName: properties: pkgs.writeTextDir fileName
+                    (concatStringsSep "\n" (mapAttrsToList (n: v: "${n}=${toString v}") properties));
+                  commonConfigFile = cfgFile "common.runtime.properties" cfg.commonConfig;
+                  configFile = cfgFile "runtime.properties" (serviceOptions.config // serviceOptions.internalConfig);
+                  extraClassPath = concatStrings (map (path: ":" + path) cfg.extraClassPaths);
+                  extraConfDir = concatStrings (map (dir: ":" + dir + "/\*") cfg.extraConfDirs);
+                in
+                ''
+                  ${cfg.package}/bin/run-java -Dlog4j.configurationFile=file:${cfg.log4j} \
```

```suggestion
                  run-java -Dlog4j.configurationFile=file:${cfg.log4j} \
```

[optional] use `run-java` from PATH instead of using absolute store path.